lib/promostandards/client.rb

```diff
@@ -64,13 +66,33 @@ def get_primary_image(product_id)
           'shar:password' => @password,
           'shar:mediaType' => 'Image',
           'shar:productId' => product_id,
-          'ns:classType' => '1006'
         },
         soap_action: 'getMediaContent'
       )
+
       media_content = response.body.dig(:get_media_content_response, :media_content_array, :media_content)
+
       if media_content.is_a? Array
-        media_content.first
+        primary_media_content = nil
+
+        PRIMARY_IMAGE_PRECEDENCE.find do |image_precendence_number|
+          primary_media_content = media_content.find do |media|
+            class_type_array = media[:class_type_array]
+
+            if class_type_array.is_a?(Hash) && image_precendence_number.is_a?(String)
```

Comment:
Not a blocker. This might be a way to reduce complexity a little bit.
```
class_type_ids = []
if class_type_array.is_a?(Hash)
  class_type_ids = [media[:class_type_array][:class_type][:class_type_id]]
elsif class_type_array.is_a?(Array)
  class_type_ids = media[:class_type_array].map { |item| item[:class_type][:class_type_id] }
end
if image_precendence_number.is_a?(Array)
  (class_type_ids & image_precendence_number).any?
else
  class_type_ids.include?(image_precendence_number)
end
```